From 12f262db0bb09d095b31522e3b37cae0bf133655 Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Mon, 31 Mar 2014 17:31:06 +0200
Subject: [PATCH 1/2] facebook: Update README

Facebook changed the layout of their OAuth2 developer documentation
website. Add new URLs with information relevant to us.

Fixes: https://bugzilla.gnome.org/726609
---
 README | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/README b/README
index 2cb2cb7..97464af 100644
--- a/README
+++ b/README
@@ -5,7 +5,11 @@ GNOME Online Accounts - Single sign-on framework for GNOME
 Facebook
 --------
 
-OAuth 2.0: https://developers.facebook.com/docs/authentication/
+OAuth 2.0:
+https://developers.facebook.com/docs/authentication/
+https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow
+https://developers.facebook.com/docs/reference/dialogs/oauth/
+
 Scopes: https://developers.facebook.com/docs/authentication/permissions/
 
 Notes:
-- 
1.8.5.3


From 0673724e0af4d7f619dbee020e0feddca730a967 Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Mon, 31 Mar 2014 18:06:58 +0200
Subject: [PATCH 2/2] facebook: Update the code to request a compact web UI

Earlier we were using m.facebook.com as the host in the authorization
URI. That does not work anymore. Now we need to use www.facebook.com
and display=popup.

According to:
https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow
https://developers.facebook.com/docs/reference/dialogs/oauth/

Fixes: https://bugzilla.gnome.org/726609
---
 src/goabackend/goafacebookprovider.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/goabackend/goafacebookprovider.c b/src/goabackend/goafacebookprovider.c
index 481b6aa..8d0814a 100644
--- a/src/goabackend/goafacebookprovider.c
+++ b/src/goabackend/goafacebookprovider.c
@@ -1,6 +1,6 @@
 /* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
 /*
- * Copyright (C) 2011, 2012, 2013 Red Hat, Inc.
+ * Copyright (C) 2011, 2012, 2013, 2014 Red Hat, Inc.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -104,6 +104,7 @@ build_authorization_uri (GoaOAuth2Provider  *provider,
 
   uri = g_strdup_printf ("%s"
                           "?response_type=token"
+                          "&display=popup"
                           "&redirect_uri=%s"
                           "&client_id=%s"
                           "&scope=%s",
@@ -117,7 +118,7 @@ build_authorization_uri (GoaOAuth2Provider  *provider,
 static const gchar *
 get_authorization_uri (GoaOAuth2Provider *provider)
 {
-  return "https://m.facebook.com/dialog/oauth";
+  return "https://www.facebook.com/dialog/oauth";
 }
 
 static const gchar *
-- 
1.8.5.3

